applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -842,47 +900,54 @@ static struct bt_conn_cb conn_callbacks = {
 
 };
 
-static int iso_stream_send(uint8_t const *const data, size_t size, uint8_t iso_chan_idx)
+static int iso_stream_send(uint8_t const *const data, size_t size, struct le_audio_headset headset)
 {
 	int ret;
-	static bool wrn_printed[CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SNK_COUNT];
 	struct net_buf *buf;
 
-	if (audio_streams[iso_chan_idx].ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
-		LOG_DBG("Channel not connected %d", iso_chan_idx);
+	if (headset.sink_stream->ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
+		LOG_DBG("%s channel not connected", headset.ch_name);
 		return 0;
 	}
 
-	if (is_iso_buffer_full(iso_chan_idx)) {
-		if (!wrn_printed[iso_chan_idx]) {
-			LOG_WRN("HCI ISO TX overrun on ch %d - Single print", iso_chan_idx);
-			wrn_printed[iso_chan_idx] = true;
+	/* net_buf_alloc allocates buffers for APP->NET transfer over HCI RPMsg,
+	 * but when these buffers are released it is not guaranteed that the
+	 * data has actually been sent. The data might be queued on the NET core,
+	 * and this can cause delays in the audio.
+	 * When stream_sent_cb() is called the data has been sent.
+	 * Data will be discarded if allocation becomes too high, to avoid audio delays.
+	 * If the NET and APP core operates in clock sync, discarding should not occur.
+	 */
+	if (atomic_get(&headset.iso_tx_pool_alloc) >= HCI_ISO_BUF_ALLOC_PER_CHAN) {
+		if (!headset.wrn_printed) {
+			LOG_WRN("HCI ISO TX overrun on %s ch - Single print", headset.ch_name);
+			headset.wrn_printed = true;
 		}
 
 		return -ENOMEM;
 	}
 
-	wrn_printed[iso_chan_idx] = false;
+	headset.wrn_printed = false;
```

Comment:
Suggest rename to `hci_overrun` or `tx_pool_overflow`

Reply:
Maybe hci_wrn_printed then?